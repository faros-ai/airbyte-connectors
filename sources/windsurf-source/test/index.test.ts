describe('index', () => {
  test('ok?', async () => {
    expect('Ok').toEqual('Ok');
  });
});
