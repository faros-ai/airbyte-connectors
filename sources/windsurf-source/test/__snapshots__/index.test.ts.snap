// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index check connection - invalid config 1`] = `
AirbyteConnectionStatusMessage {
  "connectionStatus": {
    "message": "Failed to connect to Windsurf API. Please check your service key.: Invalid service key",
    "status": "FAILED",
  },
  "type": "CONNECTION_STATUS",
}
`;

exports[`index check connection - valid config 1`] = `
AirbyteConnectionStatusMessage {
  "connectionStatus": {
    "status": "SUCCEEDED",
  },
  "type": "CONNECTION_STATUS",
}
`;

exports[`index streams - autocomplete analytics 1`] = `
[
  {
    "date": "2024-01-15",
    "email": "john@company.com",
    "ide": "vscode",
    "language": "typescript",
    "num_acceptances": 45,
    "num_lines_accepted": 280,
  },
  {
    "date": "2024-01-15",
    "email": "jane@company.com",
    "ide": "vscode",
    "language": "typescript",
    "num_acceptances": 45,
    "num_lines_accepted": 280,
  },
]
`;

exports[`index streams - cascade lines 1`] = `
[
  {
    "day": "2024-01-15",
    "email": "john@company.com",
    "linesAccepted": 80,
    "linesSuggested": 150,
  },
  {
    "day": "2024-01-16",
    "email": "john@company.com",
    "linesAccepted": 120,
    "linesSuggested": 200,
  },
  {
    "day": "2024-01-15",
    "email": "jane@company.com",
    "linesAccepted": 80,
    "linesSuggested": 150,
  },
  {
    "day": "2024-01-16",
    "email": "jane@company.com",
    "linesAccepted": 120,
    "linesSuggested": 200,
  },
]
`;

exports[`index streams - cascade runs 1`] = `
[
  {
    "cascadeId": "cascade_123",
    "day": "2024-01-15",
    "email": "john@company.com",
    "messagesSent": 5,
    "mode": "chat",
    "model": "gpt-4",
    "promptsUsed": 3,
  },
  {
    "cascadeId": "cascade_456",
    "day": "2024-01-16",
    "email": "john@company.com",
    "messagesSent": 8,
    "mode": "completion",
    "model": "claude-3",
    "promptsUsed": 6,
  },
  {
    "cascadeId": "cascade_123",
    "day": "2024-01-15",
    "email": "jane@company.com",
    "messagesSent": 5,
    "mode": "chat",
    "model": "gpt-4",
    "promptsUsed": 3,
  },
  {
    "cascadeId": "cascade_456",
    "day": "2024-01-16",
    "email": "jane@company.com",
    "messagesSent": 8,
    "mode": "completion",
    "model": "claude-3",
    "promptsUsed": 6,
  },
]
`;

exports[`index streams - user page analytics 1`] = `
[
  {
    "activeDays": 15,
    "apiKey": "hashed_api_key_123",
    "disableCodeium": false,
    "email": "john@company.com",
    "lastAutocompleteUsageTime": "2024-03-18T09:45:00Z",
    "lastChatUsageTime": "2024-03-18T10:15:00Z",
    "lastCommandUsageTime": "2024-03-18T10:20:00Z",
    "lastUpdateTime": "2024-03-18T10:30:00Z",
    "name": "John Developer",
    "signupTime": "2024-01-15T08:00:00Z",
  },
  {
    "activeDays": 22,
    "apiKey": "hashed_api_key_456",
    "email": "jane@company.com",
    "lastAutocompleteUsageTime": "2024-03-17T14:00:00Z",
    "lastChatUsageTime": "2024-03-17T13:30:00Z",
    "lastCommandUsageTime": "2024-03-17T14:10:00Z",
    "lastUpdateTime": "2024-03-17T14:20:00Z",
    "name": "Jane Engineer",
    "signupTime": "2024-01-10T12:00:00Z",
  },
]
`;
